docs/vnext/openapi.yaml

```diff
@@ -0,0 +1,405 @@
+openapi: 3.0.3
+info:
+  title: Azure Deployments Extensibility Provider API
+  description: |-
+    This is the OpenAPI 3.0 based specification for Azure Deployments Extensibility Provider API.
+  version: 2.0.0
+externalDocs:
+  description: Find out more about OpenAPI.
+  url: https://www.openapis.org/
+servers:
+  - url: https://{providerApiEndpoint}
+    variables:
+      providerApiEndpoint:
+        default: 'api.example-provider.com'
+        description: Extensibility provider API endpoint.
+tags:
+  - name: Resource
+    description: Resource API
+  - name: ResourceOperation
+    description: Resource Operation API
+paths:
+  /{providerVersion}/resources/reference:
+    post:
+      tags:
+        - Resource
+      summary: Create a resource reference
+      description: |
+        Generate a reference to a resource comprising a reference ID, regardless the existence of the resource.
+      operationId: createResourceReference
+      parameters:
+        - $ref: '#/components/parameters/HeaderAcceptLanguage'
+        - $ref: '#/components/parameters/HeaderRequestId'
+        - $ref: '#/components/parameters/HeaderCorrelationId'
+        - $ref: '#/components/parameters/HeaderTraceParent'
+        - $ref: '#/components/parameters/HeaderTraceState'
+        - $ref: '#/components/parameters/ProviderVersion'
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Resource'
+        required: true
+      responses:
+        '200':
+          description: Successful operation
+          headers: 
+            Date:
+              $ref: '#/components/headers/Date'
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ResourceReference'
+        '400':

```

Comment:
Technically many other HTTP status codes are possible to be returned (500, 503). Are we only including the intentional ones?

Reply:
Good question. I can add a `default` response to cover all other cases.